test/extended/builds/start.go

```diff
@@ -248,4 +248,24 @@ var _ = g.Describe("[builds][Slow] starting a build using CLI", func() {
 
 	})
 
+	g.Describe("Setting build-args on Docker builds", func() {
+		g.It("Should accept build args that are specified in the Dockerfile", func() {
+			g.By("starting the build with --build-args flag")
+			br, _ := exutil.StartBuildAndWait(oc, "sample-build", fmt.Sprintf("--from-dir=%s", exampleBuild), "--build-args=foo=cheese")
+			br.AssertSuccess()
+			buildLog, err := br.Logs()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By(fmt.Sprintf("verifying the build output contains the changes.", br.BuildPath))
+			o.Expect(buildLog).To(o.ContainSubstring("cheese"))
+		})
+		g.It("Should fail on non-existent build-arg", func() {
+			g.By("starting the build with --build-args flag")
+			br, _ := exutil.StartBuildAndWait(oc, "sample-build", fmt.Sprintf("--from-dir=%s", exampleBuild), "--build-args=bar=foo")
+			br.AssertSuccess()
+			buildLog, err := br.Logs()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By(fmt.Sprintf("verifying the build failed due to Docker.", br.BuildPath))
+			o.Expect(buildLog).To(o.ContainSubstring("FOO"))
```

Comment:
isn't the successful build also going to contain "FOO"?  seems like we need a more stringent expected text to check for.